util/util.go

```diff
@@ -384,13 +384,16 @@ var (
 
 // fileExistsAndNotADir - Check to see if a file exists
 // and that it is not a directory.
-func fileExistsAndNotADir(path string) bool {
+func fileExistsAndNotADir(path string) (bool, error) {
 	file, err := os.Stat(path)
 
-	if file == nil || err != nil || errors.Is(err, os.ErrNotExist) {
-		return false
+	if file == nil || err != nil {
```

Comment:
I don't think we have to to check for file ==nil either when we check the error already

Reply:
I left this in case there could be weird error, but I am fine with removing it.